base/array.jl

```diff
@@ -992,7 +993,7 @@ julia> findnext(A,3)
 """
 function findnext(A, start::Integer)
     for i = start:length(A)
-        if A[i] != 0
+        if !isequal(A[i], 0)
```

Comment:
Yes, that would fit well with this PR. Then we would need have the fallback `iszero` calll `isequal` (rather than `==`). And since `isequal(0, -0.0) == false`, we should define `iszero` on floats so that `iszero(-0.0) == true`